kong/plugins/datadog/schema.lua

```diff
@@ -74,8 +74,8 @@ return {
         type = "record",
         default = { metrics = DEFAULT_METRICS },
         fields = {
-          { host = typedefs.host({ default = "localhost" }), },
-          { port = typedefs.port({ default = 8125 }), },
+          { host = typedefs.host, },
+          { port = typedefs.port, },
```

Comment:
Changing defaults is a breaking change. Isn't it @bungle?

Reply:
@hbagdi I made a change to change the order the variables are loaded instead of changing the defaults, would it be better in this way?

Reply:
Because the field is not required, you can force the values to `null` (explicitly), then it could fallback to `ENV`. Then you don't need to change the schema.

Reply:
@bungle I already tried this and when using the KongPlugin CRD if I assign the host to null, it's removed by the plugin and it sets the default value: `localhost`, should I use a string value instead like null or nil?

Reply:
Yes, It must be that `CRD`s don't pass `null`s correctly.

Reply:
@rainest do you have any insights on `null` handling with KIC and Kong?

Reply:
Setting to an empty string (and possibly `~`, haven't tried) removed it in other configuration: https://github.com/Kong/kong/issues/8093#issuecomment-974571178

The new proposal (just checking for the presence of envvars first) probably makes more sense though, you shouldn't have to know to explicitly remove the defaults from your config.

Reply:
Please avoid explicit `null` as a differentiating parameter. Reasons:
- Even though Kong can differentiate between field not being present and an explicit `null`, a lot of tooling just can't do it.
- The difference between an explicit `null` and field not being present is obvious to us because we are the maintainers. It is far too subtle for users.
- Associating a semantic meaning to empty (`null` in this case) is best avoided. For the above reason. Use enums or other schema fields. Explicit APIs help here because we capture user's intent clearly.

@bungle What would be the downside of introducing "via_env" or some such field in the schema of this plugin?

Reply:
It looks like this is an attempted approach to fix the problems discussed in https://github.com/Kong/kong/issues/7954. I have a similar approach here: https://github.com/Kong/kong/pull/9466

Mine is also a breaking change. I can see that how the KIC/CRD interpolation works, using null/empty values leads to unexpected results in a k8s environment. 

I proposed a "via_env" style approach in my PR, would we be able to proceed in that direction?